src/Microsoft.Diagnostics.NETCore.Client/DiagnosticsServerRouter/DiagnosticsServerRouterFactory.cs

```diff
@@ -478,7 +487,17 @@ static string GetDefaultIpcServerPath()
                 unixEpoch = new DateTime(1970, 1, 1, 0, 0, 0, 0, DateTimeKind.Utc);
 #endif
                 TimeSpan diff = Process.GetCurrentProcess().StartTime.ToUniversalTime() - unixEpoch;
-                return Path.Combine(PidIpcEndpoint.IpcRootPath, $"dotnet-diagnostic-{processId}-{(long)diff.TotalSeconds}-socket");
+
+                var path = Path.Combine(PidIpcEndpoint.IpcRootPath, $"dotnet-diagnostic-{processId}-{(long)diff.TotalSeconds}-socket");
+                if (Directory.GetFiles(PidIpcEndpoint.IpcRootPath, $"dotnet-diagnostic-{processId}-*-socket").Length != 0)
+                {
+                    logger?.LogWarning($"Default IPC server path, {Path.Combine(PidIpcEndpoint.IpcRootPath, $"dotnet-diagnostic-{processId}-*-socket")}, already in use. To disable default diagnostics for dotnet-dsrouter, set COMPlus_EnableDiagnostics=0 and re-run.");
+
+                    path = Path.Combine(PidIpcEndpoint.IpcRootPath, $"dotnet-dsrouter-diagnostic-{processId}-{(long)diff.TotalSeconds}-socket");
```

Comment:
nit: Given that this is not a diagnostic port to diagnose dsrouter, could you change the name so that it is confused with being the diagnostic port for dsrouter e.g. `dotnet-dsrouter-{processId}`?

Reply:
jupp

Reply:
but if we implement connecting to named ports we don't need to fallback but can fail with error informing user to either turn of default IPC diagnostic channel or use -ipcs for an explicit named IPC channel and use --diagnostic-port and diagnostic-port-mode connect, if we decide to move down that road.

Reply:
The only way to turn off the diagnostic port server in the runtime is to set ComPlus_EnableDiagnostics=0. This is problematic for two reasons:

1. It's difficult (error prone?) to tell all users of the tool to set this environment variable for only this executable every single time you want to launch it.
1. It disables the ability to debug the process. So while you may be able to run it, you won't be able to debug it.

Ideally, there's a separate environment variable for turning off just the diagnostic port services or, even better, a runtime config switch that can be specified in the .runtimeconfig.json file that is deployed with the app.

Reply:
There is already other scenarios where you will need to run with env variables (like reverse connect), but I don't like it either since it complicates things, but if its the only option we have to disable it, that is what we have and it works on older .net versions as well. I would also love to put this in `.runtimeconfig.json`, but none of the diagnostics env's are currently included, don't know if env's in that file are set to late (needs to be set by the host or very early when launching runtime), if that's the case, then `.runtimeconfig.json` can't hold these settings.

Having a good way to turn of the default IPC listener in CoreCLR/MonoVM would be a good thing, but if we do that, we will get a dependecy on a specific .net runtime version as well when running dsrouter (but not other diagnostic tools) and that is sub optimal as well.

So sounds like we are out of good options on this, and the whole motivation to replace the default IPC channel when running dsrouter in server-server or server-client mode (client-server is unaffected by this), was to make it transparent for end users to connect and trace remote targets handled by dsrouter.

Reply:
> the whole motivation to replace the default IPC channel when running dsrouter in server-server or server-client mode (client-server is unaffected by this), was to make it transparent for end users to connect and trace remote targets handled by dsrouter.

IMO the transparency winds up being confusing because it makes it ambiguous. Is the tool diagnosing dsrouter itself (it is a .NET Core app after all) or is the tool being redirected through the proxy and diagnosing a remote process? It feels akin to using file system commands on symbolic links. For someone like myself that doesn't do it frequently I'd have to go find some docs to determine which commands indirect through the link and which ones don't.

Assuming that the underlying goal for transparency was ease-of-use I think we have alternative UX options that will remain easy without attempting to be transparent, such as [using --diagnostic-port](https://github.com/dotnet/diagnostics/pull/2374#discussion_r654223937) or perhaps some other explicit command-line option.